test/test_transform_collections_mixed.cpp

```diff
@@ -1455,3 +1461,261 @@ TEST(Transform_ClearDictionaryInsideDictionaryVsCreateDictionary)
         CHECK_EQUAL(dict->get("key2"), 2);
     });
 }
+
+TEST(Transform_UpdateClearVsUpdateClear)
+{
+    std::vector<int> timestamps{1, 2, 3, 4};
+
+    do {
+        // Baseline: property 'any' is not set to any type
+        // {id: 1, any: null}
+        TransformTestHarness h(test_context, {}, [](Obj, ColKey) {});
+
+        auto t1 = timestamps[0];
+        auto t2 = timestamps[1];
+        auto t3 = timestamps[2];
+        auto t4 = timestamps[3];
+
+        if (t1 > t2 || t3 > t4)
+            continue;
+
+        // Client 1 sets property 'any' to List and inserts one integer in the list
+        // {id: 1, any: [1]}
+        h.transaction(
+            h.client_1,
+            [](Obj obj, ColKey col_any) {
+                obj.set_collection(col_any, CollectionType::List);
+                auto l = obj.get_list<Mixed>(col_any);
+                l.add(1);
+            },
+            t1);
+
+        // Client 1 clears the list at property 'any'
+        // {id: 1, any: []}
+        h.transaction(
+            h.client_1,
+            [](Obj obj, ColKey col_any) {
+                auto l = obj.get_list<Mixed>(col_any);
+                l.clear();
+            },
+            t2);
+
+        // Client 2 sets property 'any' to List and inserts one integer in the list
+        // {id: 1, any: [2]}
+        h.transaction(
+            h.client_2,
+            [](Obj obj, ColKey col_any) {
+                obj.set_collection(col_any, CollectionType::List);
+                auto l = obj.get_list<Mixed>(col_any);
+                l.add(2);
+            },
+            t3);
+
+        // Client 2 clears the list at property 'any'
+        // {id: 1, any: []}
+        h.transaction(
+            h.client_2,
+            [](Obj obj, ColKey col_any) {
+                auto l = obj.get_list<Mixed>(col_any);
+                l.clear();
+            },
+            t4);
+
+        h.check_merge_result([&](Obj, ColKey) {});
```

Comment:
Is all we're checking here that the merge doesn't cause a bad changeset?

Reply:
yeah, and that the clients converge

Reply:
Got it. Can you just add a comment saying that these tests are just to make sure that we converge? An empty lambda here makes it seem like these tests are noops.